### PR TITLE
Exercise 5

### PR DESCRIPTION
**Pull request by cjarv4**

### CODE REVIEW

File: Dockerfile
Comment (Jonesey13) [inline Dockerfile]: This implies that this isn't a multistage build

File: Dockerfile
Comment (Jonesey13) [inline Dockerfile]: Why are these lines commented out?

File: README.md
Comment (Jonesey13) [inline README.md]: There is no `--target` here

File: README.md
Comment (Jonesey13) [inline README.md]: Assuming this for development, where is the bind mount to local files (for hot reloading)?

File: docker-compose.yml
Comment (Jonesey13) [inline docker-compose.yml:1]: This file isn't mentioned in your README.md

File: entrypoint.sh
Comment (Jonesey13) [inline entrypoint.sh:1]: Containers should ideally be single purpose (no dev/production hybrid containers)

File: todo_files/app.py
Comment (Jonesey13) [inline todo_files/app.py:9]: Why do we have code changes in this PR?

File: README.md
Comment (Jonesey13) [inline README.md:50]: README is missing instructions for setting up the `.env` file.

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
You've got most of this right, however there are few things preventing this passing review:
- The application doesn't run out of the box as the python files have capitalisation issues (e.g. `Card.py` which breaks imports
- The dockerfile isn't really a multistage build (at least not in spirit) and how to target each stage isn't documented in the README.md
- The docker run instructions does not include a bind mount (so no hot reloading of code in dev mode)
- There are no detailed instructions for setting up the `.env` file in `README.md`, esp regarding the trello credentials

File: Dockerfile
Comment (cjarv4) [reply Dockerfile]: Have removed these and uncommented the mulitstage parts below

File: Dockerfile
Comment (cjarv4) [reply Dockerfile]: uncommented

File: README.md
Comment (cjarv4) [reply README.md]: added full docker build code here

File: README.md
Comment (cjarv4) [reply README.md]: bind mount was held in the makefile. ive updated the readme to have the full docker build scripts

File: README.md
Comment (JackMeadDev) [inline README.md:55]: Minor, but it would be nice to mention the dependency on make explicitly as we're adding that as the only documented way to run the prod server

**Review by JackMeadDev — COMMENTED** [review]
It looks like this PR has already been merged; in future could you aim to leave the PR open until it's approved? If you need to build upon existing code, you can raise your newer branches off existing feature branches. 

Nice work fixing up most of the issues highlighted above, but the application still doesn't run out of the box either within docker or on my machine due to the capitalisation issues that Alex mentioned - e.g. app.py calls `import card` but the file is `Card.py` and consequently python fails to find it.

If it's not clear what the problem is there, feel free to reach out on slack or organise a support call

**Comment by JackMeadDev** [comment]
Thanks for making those changes, the case difference is likely due to a rename at some stage - git doesn't necessarily pick that up as a change, and so leaves the original casing. The app is now running in docker using the commands in the make file, so happy to sign this one off.

It's worth noting that your docker-compose file isn't currently running, because the command specified is additional to the entrypoint, so rather than `poetry run flask run --host 0.0.0.0` you'll just need to specify `--host 0.0.0.0`

**Pull request merged**
